### PR TITLE
hotfix/CVSB-2977

### PR DESCRIPTION
**Pull request by danieliordache**

### CODE REVIEW

**Comment by danieliordache** [comment]
Also closes CVSB-2979 and CVSB-3227

File: src/pages/testing/test-creation/test-create/test-create.ts [hunk 1]
```diff
@@ -158,7 +157,10 @@ export class TestCreatePage implements OnInit {
   }
 
   onOdometer(index: number) {
-    this.navCtrl.push('OdometerReadingPage', {vehicle: this.testData.vehicles[index]});
+    const MODAL = this.modalCtrl.create('OdometerReadingPage', {
```
Comment (adpetrescu) [inline src/pages/testing/test-creation/test-create/test-create.ts]: can we please move the page names in an enum as we go?

File: src/providers/global/state-reforming.service.ts [hunk 2]
```diff
@@ -1,24 +1,49 @@
...
 
 @Injectable()
 export class StateReformingService {
-  stateHistory = [];
 
-  constructor(public storageService: StorageService) {
+  constructor(public storageService: StorageService, private visitService: VisitService) {
   }
 
   saveNavStack(nav) {
```
Comment (adpetrescu) [inline src/providers/global/state-reforming.service.ts]: can you please add test coverage to the `saveNavStack` and `onTestReview` methods?

File: src/providers/global/state-reforming.service.ts [hunk 3]
```diff
@@ -1,24 +1,49 @@
...
-      const view = nav.getByIndex(i);
-      this.stateHistory.push({
-        page: view.name,
-        params: view.data
-      });
+    if (this.visitService.caching == 'true') {
+      let stateHistory: StateHistoryModel[] = [];
+      for (let i = 0; i < nav.length(); i++) {
+        const view = nav.getByIndex(i);
+        if (view.name == 'TestTypesListPage') continue;
```
Comment (adpetrescu) [inline src/providers/global/state-reforming.service.ts]: can we please move the page names in an enum as we go?

File: src/pages/testing/test-creation/test-create/test-create.ts (hunk 1 above)
Comment (mgherceanu) [reply src/pages/testing/test-creation/test-create/test-create.ts]: PAGE_NAMES.DONE

File: src/providers/global/state-reforming.service.ts (hunk 3 above)
Comment (mgherceanu) [reply src/providers/global/state-reforming.service.ts]: PAGE_NAMES.DONE

**Pull request merged**